tee-worker/core/parentchain/indirect-calls-executor/src/lib.rs

```diff
@@ -248,7 +248,10 @@ where
 		})??;
 
 		let root: H256 = merkle_root::<Keccak256, _>(extrinsics);
-		Ok(OpaqueCall::from_tuple(&(call, block_hash, block_number, root)))
+		let parentchain_block_number: ParentchainBlockNumber =
+			block_number.try_into().map_err(|_| Error::ConvertParentchainBlockNumber)?;
+		let block: u32 = parentchain_block_number;
```

Comment:
Does `codec::Compact(block_number)` work? As `<<ParentchainBlock as ParentchainBlockTrait>::Header as Header>::Number` should implement `AtLeast32BitUnsigned`. 

If it doesn't work - `codec::Compact(parentchain_block_number)` should be enough, we don't need the extra `block` variable here